solidity/contracts/liquidity-protection/LiquidityProtectionStore.sol

```diff
@@ -527,28 +545,120 @@ contract LiquidityProtectionStore is ILiquidityProtectionStore, Owned, TokenHand
         emit SystemBalanceUpdated(_token, prevAmount, newAmount);
     }
 
-    /**
-     * @dev returns the total protected pool token amount for a given pool
-     *
-     * @param _poolToken   pool token address
-     * @return total protected amount
-     */
-    function totalProtectedPoolAmount(IDSToken _poolToken) external view returns (uint256) {
-        return totalProtectedPoolAmounts[_poolToken];
+    function toUint128(uint256 _amount) private pure returns (uint128) {
+        require(_amount <= MAX_UINT128, "ERR_AMOUNT_TOO_HIGH");
+        return uint128(_amount);
     }
 
-    /**
-     * @dev returns the total protected reserve amount for a given pool
-     *
-     * @param _poolToken       pool token address
-     * @param _reserveToken    reserve token address
-     * @return total protected amount
-     */
-    function totalProtectedReserveAmount(IDSToken _poolToken, IERC20Token _reserveToken)
-        external
-        view
-        returns (uint256)
-    {
-        return totalProtectedReserveAmounts[_poolToken][_reserveToken];
+    function encodeReserveRateInfo(
```

Comment:
need to add a public utility function for decoding reserve rate info to ease on integrations

Reply:
I don't understand this requirement. The reserve-rate info is a field in a private mapping.
Function `position(uint256 _id)` already returns its decoded values (N, D and timestamp).
There is no way to access this field from outside of the contract, so you won't have any proper input to pass to such utility function.